great_expectations/experimental/datasources/sql_datasource.py

```diff
@@ -316,12 +322,28 @@ class SQLDatasource(Datasource):
     connection_string: str
     assets: Dict[str, TableAsset] = {}
 
-    @property
-    def execution_engine_type(self) -> Type[ExecutionEngine]:
-        """Returns the default execution engine type."""
-        from great_expectations.execution_engine import SqlAlchemyExecutionEngine
+    # non-field private attrs
+    _engine: Engine = pydantic.PrivateAttr()
 
-        return SqlAlchemyExecutionEngine
+    @property
+    def engine(self) -> Engine:
+        return self._engine
+
+    class Config:
+        arbitrary_types_allowed = True
+        extra = pydantic.Extra.forbid
```

Comment:
I think the configs are combined so you shouldn't need to set `extra` here.
```suggestion
```

Reply:
Actually, why do we need  `arbitrary_types_allowed = True` at all?
`_engine` is a `PrivateAttr` so doesn't `pydantic` just ignore it?

Reply:
I removed it with the recent changes. I think this was because I didn't always have `_engine` as a `PrivateAttr`.

Reply:
✅